tests/sentry/shared_integrations/client/test_base.py

```diff
@@ -34,7 +34,7 @@ def test_finalize_request(self, mock_finalize_request: MagicMock) -> None:
         assert get_response.call_count == 0
         self.api_client.get("https://example.com/get")
         assert mock_finalize_request.called
-        assert get_response.call_count == 1
+        assert get_response.call_count == 1  # type:ignore[unreachable]
```

Comment:
we shouldn't be introducing new ignores -- especially unreachable (which means anything after this becomes completely unchecked!)  sometimes you'll need to add small helpers to avoid the narrowing (though I'm not sure why narrowing is involved at all?)